### PR TITLE
Noahcaldwell/halloffame

### PR DESCRIPTION
**Pull request by Debob123**
it can update halloffame data

### CODE REVIEW

File: .DS_Store
Comment (ickoxii) [inline .DS_Store:1]: Add .DS_Store to the `.gitignore` file. We don't want to track this. After you add .DS_Store to the gitignore, run `git rm --cached \`find . -name \.DS_Store\`` to stop tracking these files

File: .Rhistory
Comment (ickoxii) [inline .Rhistory:1]: likewise, add `.Rhistory` to the gitignore and remove tracking with `git rm --cached .Rhistory`

File: dbSetup/services/halloffame_csv_service.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    print("Updating halloffame table")
+    csv_file_path = get_csv_path("HallOfFame.csv")
+
+    if len(csv_file_path) == 0:
+        print("Error: HallOfFame.csv not found")
+        return
+
+    # Process the CSV file
+    try:
+        print(update_fielding_from_csv(csv_file_path))
```
Comment (cateburglar) [inline dbSetup/services/halloffame_csv_service.py:18]: LOL you should change this function name and the function definition to the correct one. (update_halloffame_from_csv). its nbd but just for consistency

File: .python-version
Comment (ickoxii) [inline .python-version:1]: After pulling in main, you just have to run `git rm --cached .python-version`

**Comment by cateburglar** [comment]
looks good, just a note that u dont have the implementation of batting stats view? like the actual calculation of the stats and "reading in" of data. might be coming in a different PR just wanted to make a note.

**Comment by ickoxii** [comment]
> looks good, just a note that u dont have the implementation of batting stats view? like the actual calculation of the stats and "reading in" of data. might be coming in a different PR just wanted to make a note.

Looks like he does have a batting stats view branch but its not in a pr yet

**Comment by ickoxii** [comment]
***DO NOT MERGE THIS YET***
Lots of merge conflicts in `apps` to go through still

**Comment by ickoxii** [comment]
> _**DO NOT MERGE THIS YET**_ Lots of merge conflicts in `apps` to go through still

Merge conflicts fixed. good to go

File: dbSetup/models/tables/__init__.py [hunk 2]
```diff
@@ -503,7 +505,22 @@ class Fielding(Base):
     f_WP = Column(SmallInteger, nullable=True)
     f_SB = Column(SmallInteger, nullable=True)
     f_CS = Column(SmallInteger, nullable=True)
-    f_ZR = Column(Float, nullable=True)
+    f_ZR = Column(SmallInteger, nullable=True)
```
Comment (brendonnewt) [inline dbSetup/models/tables/__init__.py:508]: This needs to be a double.

File: dbSetup/models/tables/__init__.py [hunk 3]
```diff
@@ -503,7 +505,22 @@ class Fielding(Base):
...
+    halloffame_ID = Column(Integer, primary_key=True, nullable=False)
+    playerID = Column(String(9), ForeignKey("people.playerID"), nullable=False)
+    yearID = Column(SmallInteger, nullable=False)
+    votedBy = Column(String(64), nullable=False)
+    ballots = Column(SmallInteger, nullable=True)
+    needed = Column(SmallInteger, nullable=True)
+    votes = Column(SmallInteger, nullable=True)
+    inducted = Column(String(1), nullable=True)
+    category = Column(String(20), nullable=True)
+    note = Column(String(25), nullable=True)
```
Comment (brendonnewt) [inline dbSetup/models/tables/__init__.py:523]: This needs to be 255.

File: dbSetup/services/halloffame_csv_service.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                )
+                .first()
+            )
+            if existing_entry:
+                updated_rows += 1
+            
+            else:
+                new_rows += 1
+
+            session.merge(halloffame_record)
```
Comment (brendonnewt) [inline dbSetup/services/halloffame_csv_service.py:71]: We figured out we can't use session.merge for things like this. We have to manually check if it exists and modify it's fields.

**Review by brendonnewt — CHANGES_REQUESTED** [review]

File: dbSetup/models/tables/__init__.py (hunk 3 above)
Comment (ickoxii) [reply dbSetup/models/tables/__init__.py:523]: the note attribute in our database is a varchar(25). trying to read any more will throw exceptions

File: dbSetup/models/tables/__init__.py (hunk 2 above)
Comment (ickoxii) [reply dbSetup/models/tables/__init__.py:508]: i tried using Double but i dont think thats a valid attribute. sqlalchemy uses the Float keyword

**Pull request closed without merging**